java/integTest/src/test/java/glide/modules/VectorSearchTests.java

```diff
@@ -825,4 +827,111 @@ public void ft_aliasadd_aliasdel_aliasupdate() {
         assertInstanceOf(RequestException.class, exception.getCause());
         assertTrue(exception.getMessage().contains("Index does not exist"));
     }
+
+    @SneakyThrows
+    @Test
+    public void ft_explain() {
+
+        String indexName = UUID.randomUUID().toString();
+        createIndexHelper(indexName);
+
+        // search query containing numeric field.
+        String query = "@price:[0 10]";
+        String result = FT.explain(client, indexName, query).get();
+        assertTrue(result.contains("price"));
+        assertTrue(result.contains("0"));
+        assertTrue(result.contains("10"));
+
+        GlideString resultGS = FT.explain(client, gs(indexName), gs(query)).get();
+        assertTrue((resultGS).toString().contains("price"));
+        assertTrue((resultGS).toString().contains("0"));
+        assertTrue((resultGS).toString().contains("10"));
+
+        // search query that returns all data.
+        resultGS = FT.explain(client, gs(indexName), gs("*")).get();
+        result = resultGS.toString();
+        assertTrue(result.contains("*"));
+
+        assertEquals(OK, FT.dropindex(client, indexName).get());
+
+        // missing index throws an error.
+        var exception =
+                assertThrows(
+                        ExecutionException.class,
+                        () -> FT.explain(client, UUID.randomUUID().toString(), "*").get());
+        assertInstanceOf(RequestException.class, exception.getCause());
+        assertTrue(exception.getMessage().contains("Index not found"));
+    }
+
+    @SneakyThrows
+    @Test
+    public void ft_explaincli() {
+
+        String indexName = UUID.randomUUID().toString();
+        createIndexHelper(indexName);
+
+        // search query containing numeric field.
+        String query = "@price:[0 10]";
+        String result[] = FT.explaincli(client, indexName, query).get();
+        List<String> resultList = new ArrayList<>();
```

Comment:
As an interesting exercise, you can use Stream() to map result to a result list.  Something like: 
```
List<String> trimmedResult = result.stream().map(::trim).collect(Collection.toList());
```

Reply:
unfortunately streams only work on lists

Reply:
Array.stream() works on arrays.  I'll point you to an example, and IntelliJ has a lot of hints to help create streams. 

https://github.com/valkey-io/valkey-glide/blob/main/java/client/src/main/java/glide/utils/ArrayTransformUtils.java#L162-L164